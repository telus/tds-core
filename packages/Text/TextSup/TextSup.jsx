import React from 'react'
import PropTypes from 'prop-types'

import safeRest from '../../../shared/utils/safeRest'

import styles from '../../../shared/styles/Typography/Typography.modules.scss'

/**
 * Superscript text an as HTML `<sup>` element.
 *
 * _This component can only be accessed as a name-spaced component: `Text.Sup`._
 */
const TextSup = ({ children, ...rest }) => (
  <sup {...safeRest(rest)} className={styles.sup}>
    {children}
  </sup>
)

TextSup.propTypes = {
  /**
   * The text.
   */
  children: PropTypes.node.isRequired,
}

TextSup.displayName = 'Text.Sup'

export default TextSup
