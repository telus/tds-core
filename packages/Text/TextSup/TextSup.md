Only use `Text.Sup` as a child of `Paragraph` or `Text`. Use [`Heading.Sup`](#headingsup) for superscript within headings.

```
<Text>
  Enjoy buffer-free streaming any time of day<Text.Sup>2</Text.Sup>.
</Text>
```
