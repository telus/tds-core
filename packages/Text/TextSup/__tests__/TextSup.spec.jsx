import React from 'react'
import { shallow } from 'enzyme'

import TextSup from '../TextSup'

describe('TextSup', () => {
  const doShallow = props => shallow(<TextSup {...props}>Some content</TextSup>)

  it('renders', () => {
    const textSup = doShallow()

    expect(textSup).toMatchSnapshot()
  })

  it('renders an HTML sub tag', () => {
    const textSup = doShallow()

    expect(textSup).toHaveDisplayName('sup')
  })

  it('passes additional attributes to the sub element', () => {
    const textSup = doShallow({ id: 'the-sup-text' })

    expect(textSup).toHaveProp('id', 'the-sup-text')
  })

  it('does not allow custom CSS', () => {
    const textSup = doShallow({ className: 'my-custom-class', style: { color: 'hotpink' } })

    expect(textSup).not.toHaveProp('className', 'my-custom-class')
    expect(textSup).not.toHaveProp('style')
  })
})
