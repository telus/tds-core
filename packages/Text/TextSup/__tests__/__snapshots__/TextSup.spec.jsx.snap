// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextSup renders 1`] = `
<sup
  className="sup"
>
  Some content
</sup>
`;
